core/src/main/scala/org/apache/spark/deploy/history/HistoryServer.scala

```diff
@@ -131,6 +139,16 @@ class HistoryServer(
     attachHandler(contextHandler)
   }
 
+  def handleDeleteLogRequest(request: HttpServletRequest): Unit = {
+    if (deleteLogsEnabled && securityManager.checkModifyPermissions(request.getRemoteUser)) {
```

Comment:
I'm not sure that modify permissions is correct here, should we restrict this further to admin?